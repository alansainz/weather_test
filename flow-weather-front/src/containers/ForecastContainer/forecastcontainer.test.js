import React from 'react';
import ReactDOM from 'react-dom';
import { ForecastContainer } from './index.js'
import { shallow } from 'enzyme';
import { Provider } from "react-redux";

describe("Dropdown Component", () => {
  let wrapper;
  // our mock login function to replace the one provided by mapDispatchToProps
  const getTodayForecastRequest = jest.fn();
  const getFiveDaysForecastRequest = jest.fn();

   beforeEach(() => {
     // pass the mock function as the login prop 
     wrapper = shallow(<ForecastContainer getTodayForecastRequest={getTodayForecastRequest} getFiveDaysForecastRequest={getFiveDaysForecastRequest}/>)
   })

  it('should render with no props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the getData function on componentDidMount', () => {
    const spy = jest.spyOn(ForecastContainer.prototype, 'getData');
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
  });
  
});

