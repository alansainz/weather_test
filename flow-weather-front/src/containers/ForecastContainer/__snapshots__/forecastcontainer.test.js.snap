// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component should render with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForecastContainer
    getFiveDaysForecastRequest={
      [MockFunction] {
        "calls": Array [
          Array [
            3435910,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    getTodayForecastRequest={
      [MockFunction] {
        "calls": Array [
          Array [
            3435910,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Loading...",
    },
    "ref": null,
    "rendered": "Loading...",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Loading...",
      },
      "ref": null,
      "rendered": "Loading...",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
